packages/lib/smart-account/src/lukso/ERC725JsonCoder.ts

```diff
@@ -0,0 +1,159 @@
+        ...LSP6Schema,
+        {
+            name: 'DM3Keystore:<address>',
+            key: '0x89197c814f5df4249c18<address>',
+            keyType: 'Mapping',
+            valueType: 'bytes[CompactBytesArray]',
+            valueContent: 'Bytes',
+        },
+        {
+            name: 'DM3UserProfile:<address>',
+            key: '0xc0885b43cac3a34d9414<address>',
+            keyType: 'Mapping',
+            valueType: 'bytes[CompactBytesArray]',
+            valueContent: 'Bytes',
+        },
+    ];
+
+    private readonly erc725: ERC725;
+    constructor(_erc725: ERC725 = new ERC725(ERC725JsonCoder.schemas)) {
+        this.erc725 = _erc725;
+    }
+
+    public encodeDm3KeyStoreEntry(
+        controllerAddress: string,
+        signerPublicKey: string,
+        encryptedProfileKeys: string = '0x',
+    ): EncodeDataReturn {
+        const enc = this.erc725.encodeData(
+            [
+                {
+                    keyName: 'DM3Keystore:<address>',
+                    dynamicKeyParts: [controllerAddress],
+                    value: [
+                        ethers.utils.hexlify(
+                            ethers.utils.toUtf8Bytes(signerPublicKey),
+                        ),
+                        ethers.utils.hexlify(
+                            ethers.utils.toUtf8Bytes(encryptedProfileKeys),
+                        ),
+                    ],
+                },
+            ],
+            ERC725JsonCoder.schemas,
+        );
+
+        return enc;
+    }
+
+    public encodeDm3Profile(
+        controllerAddress: string,
+        userProfile: SignedUserProfile,
+    ): EncodeDataReturn {
+        const { profile, signature } = userProfile;
+        const { publicSigningKey, publicEncryptionKey, deliveryServices } =
+            profile;
+        const enc = this.erc725.encodeData([
+            {
+                keyName: 'DM3UserProfile:<address>',
+                dynamicKeyParts: [controllerAddress],
+                value: [
+                    ethers.utils.hexlify(
+                        ethers.utils.toUtf8Bytes(publicSigningKey),
+                    ),
+                    ethers.utils.hexlify(
+                        ethers.utils.toUtf8Bytes(publicEncryptionKey),
+                    ),
+                    ethers.utils.hexlify(
+                        ethers.utils.toUtf8Bytes(
+                            JSON.stringify(deliveryServices),
+                        ),
+                    ),
+                    ethers.utils.hexlify(ethers.utils.toUtf8Bytes(signature)),
+                ],
+            },
+        ]);
+        return enc;
+    }
+
+    public async decodeDm3Profile(
+        upAddress: string,
+    ): Promise<SignedUserProfile | undefined> {
+        const encodedUserProfile = await this.erc725.getData({
+            keyName: 'DM3UserProfile:<address>',
+            dynamicKeyParts: upAddress,
+        });
+        if (encodedUserProfile.value === null) {
```

Comment:
we could get rid of the following `as Data[]` `as String` using a typeguard like

```ts
function isStringArray(value: unknown): value is string[] {
    return Array.isArray(value) && value.every((item) => typeof item === 'string');
}


 if (
            encodedUserProfile.value === null ||
            !isStringArray(encodedUserProfile.value)
        ) {
            return undefined;
        }
```